spec/integration/knife/config_get_profile_spec.rb

```diff
@@ -22,92 +22,93 @@
   include KnifeSupport
 
   include_context "default config options"
-  include_context "with a chef repo"
 
   let(:cmd_args) { [] }
 
-  subject do
-    cmd = knife("config", "get-profile", *cmd_args, instance_filter: lambda { |instance|
-      # Fake the failsafe check because this command doesn't actually process knife.rb.
-      $__KNIFE_INTEGRATION_FAILSAFE_CHECK << " ole"
-    })
-    cmd.stdout
-  end
-
-  around do |ex|
-    # Store and reset the value of some env vars.
-    old_chef_home = ENV["CHEF_HOME"]
-    old_knife_home = ENV["KNIFE_HOME"]
-    old_home = ENV["HOME"]
-    old_wd = Dir.pwd
-    ChefConfig::PathHelper.per_tool_home_environment = "KNIFE_HOME"
-    # Clear these out because they are cached permanently.
-    ChefConfig::PathHelper.class_exec { remove_class_variable(:@@home_dir) }
-    Chef::Knife::ConfigGetProfile.reset_config_loader!
-    begin
-      ex.run
-    ensure
-      ENV["CHEF_HOME"] = old_chef_home
-      ENV["KNIFE_HOME"] = old_knife_home
-      ENV["HOME"] = old_home
-      Dir.chdir(old_wd)
-      ENV[ChefUtils.windows? ? "CD" : "PWD"] = Dir.pwd
-      ChefConfig::PathHelper.per_tool_home_environment = nil
+  when_the_repository("has a custom env") do
+    subject do
+      cmd = knife("config", "get-profile", *cmd_args, instance_filter: lambda { |instance|
+        # Fake the failsafe check because this command doesn't actually process knife.rb.
+        $__KNIFE_INTEGRATION_FAILSAFE_CHECK << " ole"
+      })
+      cmd.stdout
     end
-  end
-
-  before do
-    # Always run from the temp folder. This can't be in the `around` block above
-    # because it has to run after the before set in the "with a chef repo" shared context.
-    directory("repo")
-    Dir.chdir(path_to("repo"))
-    ENV[ChefUtils.windows? ? "CD" : "PWD"] = Dir.pwd
-    ENV["HOME"] = path_to(".")
-  end
-
-  context "with no configuration" do
-    it { is_expected.to eq "default\n" }
-  end
 
-  context "with --profile" do
-    let(:cmd_args) { %w{--profile production} }
-    it { is_expected.to eq "production\n" }
-  end
-
-  context "with an environment variable" do
     around do |ex|
-      old_chef_profile = ENV["CHEF_PROFILE"]
+      # Store and reset the value of some env vars.
+      old_chef_home = ENV["CHEF_HOME"]
+      old_knife_home = ENV["KNIFE_HOME"]
+      old_home = ENV["HOME"]
+      old_wd = Dir.pwd
+      ChefConfig::PathHelper.per_tool_home_environment = "KNIFE_HOME"
+      # Clear these out because they are cached permanently.
+      ChefConfig::PathHelper.class_exec { remove_class_variable(:@@home_dir) }
+      Chef::Knife::ConfigGetProfile.reset_config_loader!
```

Comment:
All of this state, `ENV`, `pwd`, `PathHelper`, should probably get reset in between tests but that's a dragon to slay some other time.